ui/app/components/oidc/assignment-form.js

```diff
@@ -0,0 +1,117 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import handleHasManySelection from 'core/utils/search-select-has-many';
+
+/**
+ * @module Oidc::AssignmentForm
+ * Oidc::AssignmentForm components are used to display the create view for OIDC providers assignments.
+ *
+ * @example
+ * ```js
+ * <Oidc::AssignmentForm @model={this.model}
+ * @onCancel={transition-to "vault.cluster.access.oidc.assignment"} @param1={{param1}}
+ * @onSave={transition-to "vault.cluster.access.oidc.assignments.assignment.details" this.model.name}
+ * />
+ * ```
+ * @callback onCancel
+ * @callback onSave
+ * @param {object} model - The parent's model
+ * @param {string} onCancel - callback triggered when cancel button is clicked
+ * @param {string} onSave - callback triggered when save button is clicked
+ */
+
+export default class OidcAssignmentFormComponent extends Component {
+  @service store;
+  @service flashMessages;
+
+  targetTypes = [
+    { label: 'Entity', type: 'identity/entity', key: 'entity_ids' },
+    { label: 'Group', type: 'identity/group', key: 'groups_ids' },
+  ];
+
+  @tracked modelValidations;
+  @tracked targets = [];
+
+  constructor() {
+    super(...arguments);
+    // aggregate different target array properties on model into flat list
+    this.flattenTargets();
+    // eagerly fetch identity groups and entities for use as search select options
+    this.resetTargetState();
+  }
+  // ARG TODO these functions are similar to the one used in MFA. Making a ticket to turn into a util/helper.
+  async flattenTargets() {
+    for (let { label, key } of this.targetTypes) {
+      const targetArray = await this.args.model[key];
```

Comment:
It's a bit unclear why this is an await, since most model attrs are not promises. Maybe add a note to that effect. I'm also not convinced that using `hasMany` is worth all the hassle, but I hope to be proven wrong!